src/main/java/moze_intel/projecte/emc/NormalizedSimpleStack.java

```diff
@@ -116,17 +123,23 @@ public static NormalizedSimpleStack forOreDictionary(String oreDictionaryName)
 	public static class NSSItem extends NormalizedSimpleStack {
 		public final String itemName;
 		public final int damage;
-		private NSSItem(String itemName, int damage) {
+		public final NBTTagCompound nbt;
+		private NSSItem(String itemName, int damage, NBTTagCompound nbt) {
 			this.itemName = itemName;
 			if (Item.itemRegistry.getObject(itemName) == null) {
 				throw new IllegalArgumentException("Invalid Item with itemName = " + itemName);
 			}
 			this.damage = damage;
+			this.nbt = nbt;
+		}
+		
+		private NSSItem(String itemName, int damage){
+			this(itemName, damage, null);
 		}
 
 		@Override
 		public int hashCode() {
-			return itemName.hashCode() ^ damage;
+			return itemName.hashCode() ^ damage ^ (nbt == null ? 0 : nbt.hashCode());
```

Comment:
Taking a look at how minecraft deals with nbt hashes,

NBTTagCompound's hashCode calls super(NBTBase)'s hashcode, and Map's hashcode.

NBTBase's hashcode calls this.getID(), which is abstract, so when NBTTagCompound.hashCode() is called, NBTTagCompound.getID is called. getID returns (byte)10.

According to Map's javadoc, "The hash code of a map is defined to be the sum of the hash codes of each entry in the map's entrySet() view". Taking a look at the documentation in NBTTagCompound, the Map instance's key is of type String. The String class's javadoc says "The hash code for a String object is computed as
 s[0]*31^(n-1) + s[1]*31^(n-2) + ... + s[n-1]".

None of the classes called when nbt.hashCode() is called call Object.hashCode().